File: WebApi/Interfaces/ITrHeaderService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { TrHeaderDto, CreateTrHeaderDto, UpdateTrHeaderDto } from '../Models/TrHeaderDtos';

export interface ITrHeaderService {
  getAll(): Promise<ApiResponse<TrHeaderDto[]>>;
  getById(id: number): Promise<ApiResponse<TrHeaderDto>>;
  getByBranchCode(branchCode: string): Promise<ApiResponse<TrHeaderDto[]>>;
  getByDateRange(startDate: Date, endDate: Date): Promise<ApiResponse<TrHeaderDto[]>>;
  getActive(): Promise<ApiResponse<TrHeaderDto[]>>;
  getByCustomerCode(customerCode: string): Promise<ApiResponse<TrHeaderDto[]>>;
  getByDocumentType(documentType: string): Promise<ApiResponse<TrHeaderDto[]>>;
  create(createDto: CreateTrHeaderDto): Promise<ApiResponse<TrHeaderDto>>;
  update(id: number, updateDto: UpdateTrHeaderDto): Promise<ApiResponse<TrHeaderDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}

File: Interfaces/ITrHeaderService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface ITrHeaderService
    {
        Task<ApiResponse<IEnumerable<TrHeaderDto>>> GetAllAsync();
        Task<ApiResponse<TrHeaderDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<TrHeaderDto>>> GetByBranchCodeAsync(string branchCode);
        Task<ApiResponse<IEnumerable<TrHeaderDto>>> GetByDateRangeAsync(DateTime startDate, DateTime endDate);
        Task<ApiResponse<IEnumerable<TrHeaderDto>>> GetActiveAsync();
        Task<ApiResponse<IEnumerable<TrHeaderDto>>> GetByCustomerCodeAsync(string customerCode);
        Task<ApiResponse<IEnumerable<TrHeaderDto>>> GetByDocumentTypeAsync(string documentType);
        Task<ApiResponse<TrHeaderDto>> CreateAsync(CreateTrHeaderDto createDto);
        Task<ApiResponse<TrHeaderDto>> UpdateAsync(long id, UpdateTrHeaderDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}